cmd/run.go

```diff
@@ -107,6 +109,34 @@ a commandline interface for interacting with it.`,
 			Left:  func() string { return "    init" },
 		}
 
+		// Load plugins
+		cliConf, err := getConfig(cmd.Flags())
+		if err != nil {
+			return err
+		}
+
+		plugins := []plugin.JavaScriptPlugin{}
+		for _, pluginPath := range cliConf.Plugins {
+			jsPlugin, err := lib.LoadJavaScriptPlugin(pluginPath)
```

Comment:
I'm not super happy about this way to interface with plugins. This assumes (correctly, for now) that plugins are only for JavaScript.

Ultimately, I think I want to have plugins that export many different types of plugins (say, a metrics plugin that exposes the API to add metrics, but also the metrics itself to export from K6) but I'm not 100% sure of what that looks like or how it's organized. I think this needs to be part of a design discussion where we draw boxes and arrows and the plugin writing UX needs to be heavily considered too.

Reply:
I agree that some kind of whiteboard session might be beneficial for this. Further, I really think we need to solve, or at least discuss, this before this can get merged, or we'll end up in a situation where it will get increasingly hard to change for every plugin that's written. cc @na--